include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,657 @@
-#include <range/v3/view/single.hpp>
-#include <range/v3/view/take_while.hpp>
-#include <range/v3/view/transform.hpp>
+namespace seqan3::detail
+{
+//!\brief Pop count function (calls gcc builtin function).
+template<integral_concept integral_type>
+auto pop_count(integral_type const n)
+{
+    return __builtin_popcount(n);
+}
+
+}
 
 namespace seqan3
 {
-/*!\brief Specialization values for translation frames.
- * \details The numeric values of the enums correspond to the number of frames that are represented.
- */
-enum class translation_frames
+//!\brief Specialization values for single and multiple translation frames.
+enum class translation_frames : uint8_t
 {
-    SINGLE_FRAME             = 1,
-    WITH_REVERSE_COMPLEMENT  = 2,
-    WITH_FRAME_SHIFTS        = 3,
-    SIX_FRAME                = 6
+    FWD_FRAME_0 = 1,                                    //!< The first forward frame starting at position 0
+    FWD_FRAME_1 = 1 << 1,                               //!< The second forward frame starting at position 1
+    FWD_FRAME_2 = 1 << 2,                               //!< The third forward frame starting at position 2
+    REV_FRAME_0 = 1 << 3,                               //!< The first reverse frame starting at position 0
+    REV_FRAME_1 = 1 << 4,                               //!< The second reverse frame starting at position 1
+    REV_FRAME_2 = 1 << 5,                               //!< The third reverse frame starting at position 2
+    FWD_REV_0 = FWD_FRAME_0 | REV_FRAME_0,              //!< The first forward and first reverse frame
+    FWD_REV_1 = FWD_FRAME_1 | REV_FRAME_1,              //!< The second forward and second reverse frame
+    FWD_REV_2 = FWD_FRAME_2 | REV_FRAME_2,              //!< The first third and third reverse frame
+    FWD = FWD_FRAME_0 | FWD_FRAME_1 | FWD_FRAME_2,      //!< All forward frames
+    REV = REV_FRAME_0 | REV_FRAME_1 | REV_FRAME_2,      //!< All reverse frames
+    SIX_FRAME = FWD | REV                               //!< All frames
 };
 
+//!\brief Enable bitwise operators for enum translation_frames.
+template<>
+constexpr bool add_enum_bitwise_operators<translation_frames> = true;
 }
 
 namespace seqan3::detail
 {
-/*!\brief Lambda for translation of one nucleotide triplet into a single aminoacid.
- * \details Used by view::translate_single.
- */
-auto translate = [] (auto range)
-{
-    return translate_triplet(range);
-};
 
-/*!\brief The underlying type of seqan3::view::translate_single.
+/*!\brief The return type of seqan3::view::translate_single.
+ * \implements seqan3::view_concept
+ * \implements seqan3::sized_range_concept
+ * \implements seqan3::random_access_range_concept
+ * \tparam urng_t The type of the range being translated.
+ * \param tf Translation frame to be used (translation_frames::FWD_FRAME_0,
+ * translation_frames::FWD_FRAME_1, translation_frames::FWD_FRAME_2,
+ * translation_frames::REV_FRAME_0,translation_frames::REV_FRAME_1,
+ * translation_frames::REV_FRAME_2).
```

Comment:
structs don't have params :wink: 

I think this is a leftover from the time where it was a \tparam